rosetta-source/src/main/rosetta/product-qualification-func.rosetta

```diff
@@ -9,6 +9,103 @@ import cdm.base.staticdata.asset.common.*
 
 isProduct root EconomicTerms;
 
+/*
+COMPOSABLE PRODUCT QUALIFICATION by ISDA Taxonomy
+AssetClass - BaseProduct - SubProduct - TransactionType
+
+Proposal:
+    - Create a new namespace cdm.product.qualification.isda:func for not qualifier functions used for composability. A CDM qualifieer function may support other product qualification logic (CFI?)
+    - Define a complete set of functions for ISDA Taxonomy V2 Level 1 - AssetClass
+    - Move Qualify_BaseProduct_* functions defined here to the new namespace
+*/
+
+/*
+Qualification of ISDA Taxonomy V2 Level 1 - ASSETCLASS:
+ - InterestRate : Qualify_AssetClass_InsterestRate
+ - ForeignExchange: Qualify_AssetClass_ForeignExchange
+ - Credit: Qualify_AssetClass_Credit
+ - Equity: Qualify_AssetClass_Equity
+ - Commodity: Qualify_Asset_Commodity
+*/
+func Qualify_AssetClass_InterestRate: <"Qualifies a product as having the Asset Class classification Interest Rate.">
+	inputs: economicTerms EconomicTerms (1..1)
+	output: is_product boolean (1..1)
+
+    alias underlier: economicTerms -> payout -> optionPayout only-element -> underlier
+
+	set is_product:
+	    economicTerms -> payout -> interestRatePayout only exists
+        or (
+            economicTerms -> payout -> optionPayout only exists
+            and ( underlier -> security -> securityType all = SecurityTypeEnum -> Debt
+                or if underlier exists then Qualify_AssetClass_InterestRate( underlier -> contractualProduct -> economicTerms ) = True
+            )
```

Comment:
Hi Ian, this part allows the IR qualification for option products such as Swaptions and DebtOptions. It first checks the underlier presence to avoid infinite function recursion.